test/PositionManager.ts

```diff
@@ -86,6 +88,64 @@ describe("GammaPoolFactory", function () {
         gammaPool = await GammaPool.attach(
             gammaPoolAddr // The deployed contract address
         );
+
+        const createLoanResponse = await (await posMgr.createLoan(cfmm.address, 1, owner.address, ethers.constants.MaxUint256)).wait();
+        createLoanArgs = createLoanResponse.events[1].args
+        tokenId = createLoanArgs.tokenId.toNumber()
+    });
+
+    describe("Base Functions", function () {
+        it("sendTokensCallback should FORBIDDEN", async function () {
+            await tokenA.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            await tokenB.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            
+            const sendTokensCallback =  {
+                payer: owner.address,
+                cfmm: cfmm.address,
+                protocol: 1,
+            }
+            const tokens = [tokenA.address, tokenB.address];
+            const amounts =  [10000, 10000];
+            const payee = addr1.address;
+            const data = ethers.utils.defaultAbiCoder.encode(["tuple(address payer, address cfmm, uint24 protocol)"],[sendTokensCallback]);
+            
+            const res = posMgr.sendTokensCallback(tokens, amounts, payee, data)
+            
+            await expect(res).to.be.revertedWith("FORBIDDEN");
+        })
+
+        it("sendTokensCallback should not be FORBIDDEN", async function () {
+            await tokenA.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            await tokenB.approve(posMgr.address, ethers.constants.MaxUint256);//must approve before sending tokens
+            
+            const prevBalanceOwnerA = await tokenA.balanceOf(owner.address);
+            const prevBalanceOwnerB = await tokenB.balanceOf(owner.address);
+
+            const sendTokensCallback =  {
+                payer: owner.address,
+                cfmm: cfmm.address,
+                protocol: 1,
+            }
+            const tokens = [tokenA.address, tokenB.address];
+            const amounts =  [90000, 1000];
+            const payee = addr1.address;
+            const data = ethers.utils.defaultAbiCoder.encode(["tuple(address payer, address cfmm, uint24 protocol)"],[sendTokensCallback]);
+            
+            const res = await gammaPool.testTokenCallbackFunction(posMgr.address, tokens, amounts, payee, data)
+
+            const newBalanceOwnerA = await tokenA.balanceOf(owner.address);
+            const newBalanceOwnerB = await tokenB.balanceOf(owner.address);
+
```

Comment:
Should also check the payee received the new balances

Reply:
Done